app/context.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2022 The Drivers Cooperative. All rights reserved.
+// Use of this source code is governed by a dual
+// license that can be found in the LICENSE file.
+
+package app
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"sync"
+	"sync/atomic"
+	"syscall"
+)
+
+// SignalContext interface represents the internal context.Context implementation that
+// hooks to SIGINT, SIGTERM, and SIGHUP operating system signals for graceful closing.
+//
+// SignalContext's methods can be called concurrently multiple times from any goroutine.
+type SignalContext interface {
```

Comment:
this one seems to be a copy of cfg.SignalContext, i assume cfg.SignalContext will be deleted?

Reply:
Yes. And some types from cfg.errors as well. Needs another PR just for refactoring.